.github/workflows/release.yml

```diff
@@ -1,30 +1,89 @@
     permissions:
-      contents: write
-      pull-requests: write
+      contents: write # This allows us to push to the repository and create GitHub releases
+      pull-requests: write # This allows us to create pull requests
     steps:
       - name: Checkout Repo
         uses: actions/checkout@v4
         with:
           # This makes Actions fetch all Git history so that Changesets can generate changelogs with the correct commits
           fetch-depth: 0
+          # NOTE: If we use default GITHUB_TOKEN to create the release PR, the checks on the release PR will not be triggered automatically
+          token: ${{ secrets.RELEASE_GITHUB_TOKEN || github.token }}
 
-      - uses: ./.github/actions/setup-env
+      - name: Set up the environment
+        uses: ./.github/actions/setup-env
 
       - name: Install Dependencies
         run: pnpm install --frozen-lockfile --prefer-offline
 
-      - name: Create Release Pull Request
+      - name: Create release Pull Request
+        id: pr
+        env:
+          # NOTE: If we use the default GITHUB_TOKEN to create the release PR, the checks on the release PR will not be triggered automatically
+          GITHUB_TOKEN: ${{ secrets.RELEASE_GITHUB_TOKEN || github.token }}
         uses: changesets/action@v1
+        with:
+          version: node scripts/version-alpha.mjs
+
+      - name: Check if release needs to be published
+        id: before
+        if: steps.pr.outputs.hasChangesets == 'false'
+        run: node scripts/check-release.mjs
+
+      - name: Build All Packages
+        if: steps.before.outputs.released == 'false'
+        run: pnpm run --recursive -no-bail --filter './v-next/**' --if-present build
+
+      - name: Publish All Packages (dry-run)
+        if: steps.before.outputs.released == 'false'
+        run: pnpm publish --filter "./v-next/**" -r --no-git-checks --tag next --access public --dry-run
+
+      - name: Publish All Packages
+        id: publish
+        if: steps.before.outputs.released == 'false'
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NODE_AUTH_TOKEN }}
+          NPM_CONFIG_PROVENANCE: true
+        run: pnpm publish --filter "./v-next/**" -r --no-git-checks --tag next --access public
+
+      - name: Check if release was published
+        id: after
+        if: steps.before.outputs.released == 'false'
+        run: node scripts/check-release.mjs
+
+      - name: Prepare GitHub release
+        id: release
+        if: steps.before.outputs.released == 'false' && steps.after.outputs.released == 'true'
+        run: node scripts/prepare-github-release.mjs
+
+      - name: Create GitHub Release
+        if: steps.before.outputs.released == 'false' && steps.after.outputs.released == 'true'
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_TOKEN: ${{ github.token }}
+        uses: galargh/action-gh-release@571276229e7c9e6ea18f99bad24122a4c3ec813f # https://github.com/galargh/action-gh-release/pull/1
```

Comment:
Are we able to push this to Nomic to keep us self-contained?

Reply:
Ah. What is it that drives pulling in an external task here? Is the GitHub API around releases just really cumbersome here?

Reply:
Now that I think about it, we should be able to use the API directly in our case. This action is great for a little more advanced setups. Especially those where a draft gets updated over and over again. However, in our setup, we only ever create the draft once and that's it.

I'm going to update it to use the GitHub API directly. Thanks for pointing this out. I might not be able to finish it before our meeting though.

Reply:
Cool lets do that as a follow up PR.
I will mark this as approved.